<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Cox Lab">
<meta name="dcterms.date" content="2025-10-10">

<title>Activities</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a href="./index.html" class="navbar-brand navbar-brand-logo">
    <img src="./Figures/coxlab_logo2.png" alt="" class="navbar-logo">
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html" rel="" target="">
 <span class="menu-text">HOME</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./maxquant_instructions.html" rel="" target="">
 <span class="menu-text">MaxQuant</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./perseus_instructions.html" rel="" target="">
 <span class="menu-text">Perseus</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./andromeda_instructions.html" rel="" target="">
 <span class="menu-text">Andromeda</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="./contact.html" rel="" target=""><i class="bi bi-envelope" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/cox-labs/CoxLab_Bug_Reporting" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/TheCoxLab" rel="" target=""><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/in/juergen-cox-2437ba17/" rel="" target=""><i class="bi bi-fa-linkedin fa-lg" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="-1">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#matrix-upload" id="toc-matrix-upload" class="nav-link active" data-scroll-target="#matrix-upload"><span class="header-section-number">1</span> Matrix Upload</a></li>
  <li><a href="#matrix-export" id="toc-matrix-export" class="nav-link" data-scroll-target="#matrix-export"><span class="header-section-number">2</span> Matrix Export</a></li>
  <li><a href="#matrix-analysis" id="toc-matrix-analysis" class="nav-link" data-scroll-target="#matrix-analysis"><span class="header-section-number">3</span> Matrix Analysis</a>
  <ul>
  <li><a href="#clusteringpca" id="toc-clusteringpca" class="nav-link" data-scroll-target="#clusteringpca"><span class="header-section-number">3.1</span> Clustering/PCA</a></li>
  <li><a href="#misc.-analysis" id="toc-misc.-analysis" class="nav-link" data-scroll-target="#misc.-analysis"><span class="header-section-number">3.2</span> Misc. (Analysis)</a></li>
  <li><a href="#visualization" id="toc-visualization" class="nav-link" data-scroll-target="#visualization"><span class="header-section-number">3.3</span> Visualization</a></li>
  </ul></li>
  <li><a href="#matrix-multiprocessing" id="toc-matrix-multiprocessing" class="nav-link" data-scroll-target="#matrix-multiprocessing"><span class="header-section-number">4</span> Matrix MultiProcessing</a>
  <ul>
  <li><a href="#basic-multiprocessing" id="toc-basic-multiprocessing" class="nav-link" data-scroll-target="#basic-multiprocessing"><span class="header-section-number">4.1</span> Basic (MultiProcessing)</a></li>
  <li><a href="#ci" id="toc-ci" class="nav-link" data-scroll-target="#ci"><span class="header-section-number">4.2</span> CI</a></li>
  </ul></li>
  <li><a href="#matrix-processing" id="toc-matrix-processing" class="nav-link" data-scroll-target="#matrix-processing"><span class="header-section-number">5</span> Matrix Processing</a>
  <ul>
  <li><a href="#annot.-columns" id="toc-annot.-columns" class="nav-link" data-scroll-target="#annot.-columns"><span class="header-section-number">5.1</span> Annot. columns</a></li>
  <li><a href="#annot.-rows" id="toc-annot.-rows" class="nav-link" data-scroll-target="#annot.-rows"><span class="header-section-number">5.2</span> Annot. rows</a></li>
  <li><a href="#basic-processing" id="toc-basic-processing" class="nav-link" data-scroll-target="#basic-processing"><span class="header-section-number">5.3</span> Basic (Processing)</a></li>
  <li><a href="#clustering" id="toc-clustering" class="nav-link" data-scroll-target="#clustering"><span class="header-section-number">5.4</span> Clustering</a></li>
  <li><a href="#filter-columns" id="toc-filter-columns" class="nav-link" data-scroll-target="#filter-columns"><span class="header-section-number">5.5</span> Filter columns</a></li>
  <li><a href="#filter-rows" id="toc-filter-rows" class="nav-link" data-scroll-target="#filter-rows"><span class="header-section-number">5.6</span> Filter rows</a></li>
  <li><a href="#imputation" id="toc-imputation" class="nav-link" data-scroll-target="#imputation"><span class="header-section-number">5.7</span> Imputation</a></li>
  <li><a href="#learning" id="toc-learning" class="nav-link" data-scroll-target="#learning"><span class="header-section-number">5.8</span> Learning</a></li>
  <li><a href="#modifications" id="toc-modifications" class="nav-link" data-scroll-target="#modifications"><span class="header-section-number">5.9</span> Modifications</a></li>
  <li><a href="#normalization" id="toc-normalization" class="nav-link" data-scroll-target="#normalization"><span class="header-section-number">5.10</span> Normalization</a></li>
  <li><a href="#outliers" id="toc-outliers" class="nav-link" data-scroll-target="#outliers"><span class="header-section-number">5.11</span> Outliers</a></li>
  <li><a href="#quality" id="toc-quality" class="nav-link" data-scroll-target="#quality"><span class="header-section-number">5.12</span> Quality</a></li>
  <li><a href="#rearrange" id="toc-rearrange" class="nav-link" data-scroll-target="#rearrange"><span class="header-section-number">5.13</span> Rearrange</a></li>
  <li><a href="#tests" id="toc-tests" class="nav-link" data-scroll-target="#tests"><span class="header-section-number">5.14</span> Tests</a></li>
  <li><a href="#time-series" id="toc-time-series" class="nav-link" data-scroll-target="#time-series"><span class="header-section-number">5.15</span> Time series</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Activities</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Cox Lab </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 10, 2025</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<!-- 
<ifauth @internal>//You are a user associated with MPIB, so you are presumably
using the internal version of Perseus, which has some features not available
to the general public. These internal features are visible when you view this page,
but not when an external user does so. They are marked with an asterisk (*).//</ifauth>

<ifauth @bioinformatics>//You are also associated with the bioinformatics group.
For you, developmental activities are also shown here and linked to whatever
documentation might be available, even though they are not available in the software.
They are marked with two asterisks (**).//</ifauth>
 -->
<p>Data processing in Perseus generally takes the form of loading a matrix of data and then applying a series of transformations to that matrix. These transformations are called “activities”. They are organized into a series of “Types”, and possibly a number of “Headings” under the Type.</p>
<p>You can search this page for the name of the activity of interest, or you can browse the Type or Heading of interest using the Table of Contents on the right.</p>
<section id="matrix-upload" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Matrix Upload</h1>
<ul>
<li><a href="binaryupload.html">Binary upload</a></li>
<li><a href="creategenelist.html">Create gene list</a></li>
<li><a href="createrandommatrix.html">Create random matrix</a></li>
<li><a href="genericmatrixupload.html">Generic matrix upload</a></li>
<li><a href="ngsupload.html">Next generation sequencing data upload</a></li>
<li><a href="unstructuredtxtupload.html">Raw upload</a></li>
</ul>
<!-- 
* [Affymetrix CGH](AffymetrixCghDataUpload) [[perseus:bioinformatics:activities:MatrixUpload:|]]
* [Agilent](AgilentDataUpload) [[perseus:bioinformatics:activities:MatrixUpload:|]]
* [SBML to matrix](Sbml2Matrix) [[perseus:bioinformatics:activities:MatrixUpload:|]]
 -->
</section>
<section id="matrix-export" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Matrix Export</h1>
<ul>
<li><a href="tabseparatedexport.html">Generic matrix export</a></li>
</ul>
<!-- 
* [.arff export](ArffWriter) [[perseus:bioinformatics:activities:MatrixExport:|]]
 -->
</section>
<section id="matrix-analysis" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Matrix Analysis</h1>
<!--
## ARACNe (Analysis)


* [CoExpression clustering](ARACNeClustering) [[perseus:internal:activities:MatrixAnalysis:ARACNe:|]]


<!-- 
## Clustering

* [fastcluster](FastCluster) [[perseus:bioinformatics:activities:MatrixAnalysis:Clustering:|]]
 -->
<section id="clusteringpca" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="clusteringpca"><span class="header-section-number">3.1</span> Clustering/PCA</h2>
<ul>
<li><a href="hierarchicalcluster.html">Hierarchical clustering</a></li>
<li><a href="principalcomponentanalysis.html">Principal component analysis</a></li>
<li><a href="coexpressionclustering.html">Co-expression clustering</a></li>
</ul>
<!-- 
* [Biclustering](Biclustering) [[perseus:bioinformatics:activities:MatrixAnalysis:ClusteringPCA:|]]
* [Factor analysis](FactorAnalysis) [[perseus:bioinformatics:activities:MatrixAnalysis:ClusteringPCA:|]]
* [Independent component analysis](IndependentComponentAnalysis) [[perseus:bioinformatics:activities:MatrixAnalysis:ClusteringPCA:|]]
 -->
<!-- 
# Co-expression
</ifauth>

* [Pick soft threshold](PickSoftThreshold) [[perseus:bioinformatics:activities:MatrixAnalysis:Co-expression]]
 -->
</section>
<section id="misc.-analysis" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="misc.-analysis"><span class="header-section-number">3.2</span> Misc. (Analysis)</h2>
<ul>
<li><a href="numericvenndiagram.html">Numeric venn diagram</a></li>
<li><a href="selectrowsmanually.html">Select rows manually</a></li>
<li><a href="sequencelogo.html">Sequence logos</a></li>
<li><a href="volcanoplotanalysis.html">Volcano plot</a></li>
<li><a href="hawaiiplot.html">Hawaii plot</a></li>
</ul>
<!-- 
* [Edit matrix](EditMatrix) [[perseus:bioinformatics:activities:MatrixAnalysis:Misc:|]]
* [Hawaii plot (multi-volcano)](MultiVolcanoAnalysis) [[perseus:internal:activities:MatrixAnalysis:Misc:|]]
* [Sequence correlation](SequenceCorrelation) [[perseus:bioinformatics:activities:MatrixAnalysis:Misc:|]]
 -->
</section>
<section id="visualization" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="visualization"><span class="header-section-number">3.3</span> Visualization</h2>
<ul>
<li><a href="scatterplot3danalysis.html">3D plot</a></li>
<li><a href="histogram.html">Histogram</a></li>
<li><a href="multiscatteranalysis.html">Multi scatter plot</a></li>
<li><a href="profileplot.html">Profile plot</a></li>
<li><a href="scatterplotanalysis.html">Scatter plot</a></li>
</ul>
<!-- 
* [Tree map](TreeMapAnalysis) [[perseus:bioinformatics:activities:MatrixAnalysis:Visualization:|]]**\\</ifauth>
* [Box plot](BoxPlotAnalysis) [[perseus:bioinformatics:activities:MatrixAnalysis:Visualization:|]]**\\</ifauth>
* [Charts](ChartAnalysis) [[perseus:bioinformatics:activities:MatrixAnalysis:Visualization:|]]**\\</ifauth>
 -->
</section>
</section>
<section id="matrix-multiprocessing" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Matrix MultiProcessing</h1>
<section id="basic-multiprocessing" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="basic-multiprocessing"><span class="header-section-number">4.1</span> Basic (MultiProcessing)</h2>
<ul>
<li><a href="matchingcolumnsbyname.html">Matching columns by name</a></li>
<li><a href="matchingrowsbyname.html">Matching rows by name</a></li>
<li><a href="replacestrings.html">Replace strings</a></li>
<li><a href="rowcorrelation.html">Row correlation</a></li>
<li><a href="changecolumnnames.html">Change column names</a></li>
</ul>
<!-- 
*[Venn diagram](VennMultiProcessing) [[perseus:internal:activities:MatrixMultiProcessing:Basic:|]]
-->
</section>
<section id="ci" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="ci"><span class="header-section-number">4.2</span> CI</h2>
<ul>
<li><a href="assertmatrixequal.html">Assert matrix equals</a></li>
</ul>
<!--  
## Misc. (MultiProcessing)

*[Match by chromosomal location](MatchingByGenomePosition) [[perseus:bioinformatics:activities:MatrixMultiProcessing:Misc:|]]
-->
</section>
</section>
<section id="matrix-processing" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Matrix Processing</h1>
<!-- 
## AA sequence

*[GRAVY](GravyValue) [[perseus:internal:activities:MatrixProcessing:AAsequence:|]]
*[Load sequences from fasta](LoadSequencesFromFasta) [[perseus:internal:activities:MatrixProcessing:AAsequence:|]]

## ARACNe (Processing)

*[Create adjacency matrix](ARACNeAdjacencyMatrix) [[perseus:internal:activities:MatrixProcessing:ARACNe:|]]

## Add content

* [IntAct neighborhood](IntActAnnotation) [[perseus:bioinformatics:activities:MatrixProcessing:Addcontent:|]]
* [Mouse phenotype](MousePhenotype) [[perseus:bioinformatics:activities:MatrixProcessing:Addcontent:|]]
* [Orthologs](AddOrthologs) [[perseus:bioinformatics:activities:MatrixProcessing:Addcontent:|]]
* [miRNA targets](MirnaTarget) [[perseus:bioinformatics:activities:MatrixProcessing:Addcontent:|]]
-->
<section id="annot.-columns" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="annot.-columns"><span class="header-section-number">5.1</span> Annot. columns</h2>
<ul>
<li><a href="onedimannotationenrichment.html">1D annotation enrichment</a></li>
<li><a href="twodimannotationenrichment.html">2D annotation enrichment</a></li>
<li><a href="addannotationtomatrix.html">Add annotation</a></li>
<li><a href="addannotationtomatrix.html">Annotation matrix</a></li>
<li><a href="averagecategories.html">Average categories</a></li>
<li><a href="categorycounting.html">Category counting</a></li>
<li><a href="fisherexacttestproces.html">Fisher exact test</a></li>
<li><a href="backtobaseidentifiers.html">To base identifiers</a></li>
</ul>
<!-- 
* [Add search database](AddSourceDatabase) [[perseus:internal:activities:MatrixProcessing:Annotcolumns:|]]
* [ComplexCorrelation](ComplexCorrelation) [[perseus:bioinformatics:activities:MatrixProcessing:Annotcolumns:]]
* [Count protein sequence features](AddProteinSequenceCounts) [[perseus:internal:activities:MatrixProcessing:Annotcolumns:|]]
-->
</section>
<section id="annot.-rows" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="annot.-rows"><span class="header-section-number">5.2</span> Annot. rows</h2>
<ul>
<li><a href="averagegroups.html">Average groups</a></li>
<li><a href="managecategoricalannotrow.html">Categorical annotation rows</a></li>
<li><a href="jointermsincategoricalrow.html">Join terms in categorical row</a></li>
<li><a href="managenumericalannotrow.html">Numerical annotation rows</a></li>
</ul>
<!-- 
* [Regex category row](RegexCategoryRow) [[perseus:internal:activities:MatrixProcessing:Annotrows:|]]
* [Regroup](Regroup) [[perseus:bioinformatics:activities:MatrixProcessing:Annotrows:|]]
-->
</section>
<section id="basic-processing" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="basic-processing"><span class="header-section-number">5.3</span> Basic (Processing)</h2>
<ul>
<li><a href="addnoise.html">Add noise</a></li>
<li><a href="cloneprocessing.html">Clone</a></li>
<li><a href="columncorrelations.html">Column correlation</a></li>
<li><a href="combinemaincolumns.html">Combine main columns</a></li>
<li><a href="combinebyidentifiersprocessing.html">Combine rows by identifiers</a></li>
<li><a href="densityestimationprocessing.html">Density estimation</a></li>
<li><a href="performancecurves.html">Performance curves</a></li>
<li><a href="quantiles.html">Quantiles</a></li>
<li><a href="rowcorrelations.html">Row correlation</a></li>
<li><a href="summarystatisticscolumns.html">Summary statistics (columns)</a></li>
<li><a href="summarystatisticsrows.html">Summary statistics (rows)</a></li>
<li><a href="transformationproc.html">Transform</a></li>
</ul>
<!-- 
* [Create numerical column](CreateNumericalColumn) [[perseus:bioinformatics:activities:MatrixProcessing:Basic:|]]
* [Create numerical row](CreateNumericalRow) [[perseus:bioinformatics:activities:MatrixProcessing:Basic:|]]
* [Fast Pearson column correlation](FastPearsonCorrelations) [[perseus:bioinformatics:activities:MatrixProcessing:Basic:|]]
* [Parameter dependent correlation](ParameterDependentCorrelation) [[perseus:bioinformatics:activities:MatrixProcessing:Basic:|]]
-->
</section>
<section id="clustering" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="clustering"><span class="header-section-number">5.4</span> Clustering</h2>
<ul>
<li><a href="genericclusteringproc.html">Generic clustering</a></li>
</ul>
<!-- 
* [Kernel PCA](KernelPcaProc) [[perseus:internal:activities:MatrixProcessing:Clustering:|]]
* [Kernel canonical correlation](KernelCanonicalCorrelationProc) [[perseus:internal:activities:MatrixProcessing:Clustering:|]]
* [Motif clustering](MotifClusteringProc) [[perseus:internal:activities:MatrixProcessing:Clustering:|]]
* [Peptide clustering](PeptideClusteringProc) [[perseus:internal:activities:MatrixProcessing:Clustering:|]]
-->
<!-- 
## Co-expression

* [Module Eigengenes](MatrixModuleEigengene) [[perseus:bioinformatics:activities:MatrixProcessing:Co-expression:|]]
-->
<!-- 
## Filter

* [Filter labels](FilterLabels) [[perseus:bioinformatics:activities:MatrixProcessing:Filter:|]]
-->
</section>
<section id="filter-columns" class="level2" data-number="5.5">
<h2 data-number="5.5" class="anchored" data-anchor-id="filter-columns"><span class="header-section-number">5.5</span> Filter columns</h2>
<ul>
<li><a href="filtercategoricalrow.html">Filter columns based on categorical row</a></li>
<li><a href="filtervalidvaluescolumns.html">Filter columns based on valid values</a></li>
</ul>
</section>
<section id="filter-rows" class="level2" data-number="5.6">
<h2 data-number="5.6" class="anchored" data-anchor-id="filter-rows"><span class="header-section-number">5.6</span> Filter rows</h2>
<ul>
<li><a href="filtercategoricalcolumn.html">Filter rows based on categorical column</a></li>
<li><a href="filternumericalcolumn.html">Filter rows based on numerical/main column</a></li>
<li><a href="filterrandomrows.html">Filter rows based on random sampling</a></li>
<li><a href="filtertextualcolumn.html">Filter rows based on text column</a></li>
<li><a href="filtervalidvaluesrows.html">Filter rows based on valid values</a></li>
</ul>
<!-- 
* [Filter grouped rows by valid values](GroupedValidValueFilter) [[perseus:bioinformatics:activities:MatrixProcessing:Filterrows]]
-->
<!-- 
# Genome

* [Average along chromosomes](AverageAlongChromosomes) [[perseus:internal:activities:MatrixProcessing:Genome:|]]
* [Create genome position](AddGenomePosition) [[perseus:internal:activities:MatrixProcessing:Genome:|]]
* [Genome profiling](GenomeAnalysis) [[perseus:internal:activities:MatrixProcessing:Genome:|]]
-->
</section>
<section id="imputation" class="level2" data-number="5.7">
<h2 data-number="5.7" class="anchored" data-anchor-id="imputation"><span class="header-section-number">5.7</span> Imputation</h2>
<ul>
<li><a href="replacemissingbyconstant.html">Replace missing values by constant</a></li>
<li><a href="replacemissingfromgaussian.html">Replace missing</a></li>
<li><a href="replaceimputedbynan.html">Replace imputed values by NaN</a></li>
</ul>
<!-- 
* [BPCA imputation](BpcaImputation) [[perseus:bioinformatics:activities:MatrixProcessing:Imputation:|]]
* [KNN imputation](KnnImputation) [[perseus:internal:activities:MatrixProcessing:Imputation:|]]
* [Replace missing SILAC pair ratios](ImputeSilacDoublets) [[perseus:bioinformatics:activities:MatrixProcessing:Imputation:|]]
* [Replace missing SILAC triplet ratios](ImputeSilacTriplets) [[perseus:bioinformatics:activities:MatrixProcessing:Imputation:|]]
values from normal distribution]]\\
* [Row average imputation](RowAverageImputation) [[perseus:internal:activities:MatrixProcessing:Imputation:|]]
-->
<!-- 
##  Interaction

* [Annotate from category row](InteractionDbAnnotation) [[perseus:internal:activities:MatrixProcessing:Interaction:|]]
-->
</section>
<section id="learning" class="level2" data-number="5.8">
<h2 data-number="5.8" class="anchored" data-anchor-id="learning"><span class="header-section-number">5.8</span> Learning</h2>
<ul>
<li><a href="classificationprocessing.html">Classification (cross-validation and prediction</a></li>
<li><a href="ClassificationFeatureSelection.html">Classification feature optimization</a></li>
<li><a href="ClassificationParameterOptimization.html">Classification parameter optimization</a></li>
</ul>
<!-- 
* [Regression (cross-validation and prediction](RegressionProcessing) [[perseus:internal:activities:MatrixProcessing:Learning:|)]]*\\</ifauth>
* [Regression feature optimization](RegressionFeatureSelection) [[perseus:internal:activities:MatrixProcessing:Learning:|]]*\\</ifauth>
* [Regression parameter optimization](RegressionParameterOptimization) [[perseus:internal:activities:MatrixProcessing:Learning:|]]*\\</ifauth>
-->
<!-- 
<ifauth @bioinformatics>==== Misc. (Processing) ====
</ifauth>

<ifauth @bioinformatics>• [[perseus:bioinformatics:activities:MatrixProcessing:Misc:CreateGeneLists|Create gene lists]]**\\</ifauth>
<ifauth @bioinformatics>• [[perseus:bioinformatics:activities:MatrixProcessing:Misc:AssemblePeptides|Peptide assembly]]**\\</ifauth>
<ifauth @bioinformatics>• [[perseus:bioinformatics:activities:MatrixProcessing:Misc:SmallMolMasses|Small mol masses]]**\\</ifauth>

<ifauth @bioinformatics>==== Modeling ====
</ifauth>

<ifauth @bioinformatics>• [[perseus:bioinformatics:activities:MatrixProcessing:Modeling:SimmuneSimulationResultToMqOutput|Simulation results to MQ output]]**\\</ifauth>
-->
</section>
<section id="modifications" class="level2" data-number="5.9">
<h2 data-number="5.9" class="anchored" data-anchor-id="modifications"><span class="header-section-number">5.9</span> Modifications</h2>
<ul>
<li><a href="addknownsites.html">Add known sites</a></li>
<li><a href="addlinearmotifs.html">Add linear motifs</a></li>
<li><a href="addmodificationcounts.html">Add modification counts</a></li>
<li><a href="addregulatorysites.html">Add regulatory sites</a></li>
<li><a href="addsequencefeatures.html">Add sequence features</a></li>
<li><a href="expandsitetable.html">Expand site table</a></li>
<li><a href="kinasesubstraterelations.html">Kinase-substrate relations</a></li>
<li><a href="shortenmotifs.html">Shorten motif length</a></li>
</ul>
<!-- 
* [Sequence window conversion](SequenceWindowToVector) [[perseus:bioinformatics:activities:MatrixProcessing:Modifications:|]]
-->
</section>
<section id="normalization" class="level2" data-number="5.10">
<h2 data-number="5.10" class="anchored" data-anchor-id="normalization"><span class="header-section-number">5.10</span> Normalization</h2>
<ul>
<li><a href="ClusterNorm.html">Cluster normalization</a></li>
<li><a href="divide.html">Divide</a></li>
<li><a href="normalizebycolumn.html">Modify by column</a></li>
<li><a href="rank.html">Rank</a></li>
<li><a href="scaletointerval.html">Scale to interval</a></li>
<li><a href="subtract.html">Subtract</a></li>
<li><a href="subtractrowcluster.html">Subtract row cluster</a></li>
<li><a href="unzscore.html">Un-Z-score</a></li>
<li><a href="unitvector.html">Unit vectors</a></li>
<li><a href="widthadjustment.html">Width adjustment</a></li>
<li><a href="zscore.html">Z-score</a></li>
</ul>
<!-- 
* [Background subtraction](BackgroundSubtraction) [[perseus:bioinformatics:activities:MatrixProcessing:Normalization:|]]
* [Intensity-dependent normalization](IntensityDependentNormalization2) [perseus:internal:activities:MatrixProcessing:Normalization:|]]
* [LOWESS normalization](IntensityDependentNormalization) [[perseus:bioinformatics:activities:MatrixProcessing:Normalization:|]]
* [Normalize by group](NormalizeByGroup) [[perseus:bioinformatics:activities:MatrixProcessing:Normalization:|]]
* [Optimal rotation](OptimalRotation) [[perseus:bioinformatics:activities:MatrixProcessing:Normalization:|]]
-->
</section>
<section id="outliers" class="level2" data-number="5.11">
<h2 data-number="5.11" class="anchored" data-anchor-id="outliers"><span class="header-section-number">5.11</span> Outliers</h2>
<ul>
<li><a href="significancea.html">Significance A</a></li>
<li><a href="significanceb.html">Significance B</a></li>
</ul>
<!-- 
* [Forward-reverse analysis](ForwardReverseAnalysis) [[perseus:bioinformatics:activities:MatrixProcessing:Outliers:|]]
* [Multidimensional significance](MultiDimSignificance) [[perseus:internal:activities:MatrixProcessing:Outliers:|]]
-->
</section>
<section id="quality" class="level2" data-number="5.12">
<h2 data-number="5.12" class="anchored" data-anchor-id="quality"><span class="header-section-number">5.12</span> Quality</h2>
<ul>
<li><a href="converttonan.html">Convert to NaN</a></li>
<li><a href="createqualitymatrix.html">Create quality matrix</a></li>
<li><a href="filterquality.html">Filter quality</a></li>
</ul>
</section>
<section id="rearrange" class="level2" data-number="5.13">
<h2 data-number="5.13" class="anchored" data-anchor-id="rearrange"><span class="header-section-number">5.13</span> Rearrange</h2>
<ul>
<li><a href="changecolumntype.html">Change column type</a></li>
<li><a href="combineannotations.html">Combine annotations</a></li>
<li><a href="combinecategoricalcolumns.html">Combine categorical columns</a></li>
<li><a href="convertmultinumeric.html">Convert multi-numeric column</a></li>
<li><a href="dehyphenateids.html">De-hyphenate ids</a></li>
<li><a href="duplicatecolumns.html">Duplicate columns</a></li>
<li><a href="expandmultinumeric.html">Expand multi-numeric and text columns</a></li>
<li><a href="fillcategoricalcolumn.html">Fill categorical columns</a></li>
<li><a href="processtextcolumns.html">Process text column</a></li>
<li><a href="removeemptycolumns.html">Remove empty columns</a></li>
<li><a href="renamecolumns.html">Rename columns</a></li>
<li><a href="renamecolumnsregexp.html">Rename columns [reg. ex.]</a></li>
<li><a href="reordercolumnsbynumannotationrow.html">Reorder columns by numerical annotation row</a></li>
<li><a href="reorderremoveannotrows.html">Reorder/remove annotation rows</a></li>
<li><a href="reorderremoveannotrows.html">remove columns</a></li>
<li><a href="searchtextcolumns.html">Search text column</a></li>
<li><a href="sortbycolumn.html">Sort by column</a></li>
<li><a href="transpose.html">Transpose</a></li>
<li><a href="uniquerows.html">Unique rows</a></li>
<li><a href="uniquevalues.html">Unique values</a></li>
</ul>
<!-- 
* [Collapse MaxQuant evidence File](CollapseEvidenceFile) [[perseus:internal:activities:MatrixProcessing:Rearrange:|]]
* [Column grouping to multiple rows](ColumnGroupingToMultipleRows) [[perseus:internal:activities:MatrixProcessing:Rearrange:|]]
* [Map columns to Uniprot](MapColumnsToUniprot) [[perseus:bioinformatics:activities:MatrixProcessing:Rearrange:|]]
* [Matrix to List](MatrixToList) [[perseus:internal:activities:MatrixProcessing:Rearrange:|]]
* [Rename columns from file](RenameColumnsFromFile) [[perseus:bioinformatics:activities:MatrixProcessing:Rearrange:|]]
* [Sort columns](SortColumns) [[perseus:bioinformatics:activities:MatrixProcessing:Rearrange:|]]
-->
</section>
<section id="tests" class="level2" data-number="5.14">
<h2 data-number="5.14" class="anchored" data-anchor-id="tests"><span class="header-section-number">5.14</span> Tests</h2>
<ul>
<li><a href="multiplesampletestprocessing.html">Multiple-sample tests</a></li>
<li><a href="onesampletestprocessing.html">One-sample tests</a></li>
<li><a href="posthoctests.html">Post hoc tests</a></li>
<li><a href="threewayanova.html">Three-way ANOVA</a></li>
<li><a href="twosampletestprocessing.html">Two-sample tests</a></li>
<li><a href="twowayanova.html">Two-way ANOVA</a></li>
</ul>
<!--
* [Correlation tests](CorrelationTests) [[perseus:internal:activities:MatrixProcessing:Tests:|]]
* [Four-way ANOVA](FourWayAnova) [[perseus:internal:activities:MatrixProcessing:Tests:|]]
-->
</section>
<section id="time-series" class="level2" data-number="5.15">
<h2 data-number="5.15" class="anchored" data-anchor-id="time-series"><span class="header-section-number">5.15</span> Time series</h2>
<ul>
<li><a href="cyclicannotation.html">Cyclic annotation enrichment</a></li>
<li><a href="periodicityanalysis.html">Periodicity analysis</a></li>
<li><a href="periodogram.html">Periodogram</a></li>
<li><a href="timeseriesordering.html">Time series ordering</a></li>
</ul>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">© 2023, created by and maintained by Cox lab</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>



</body></html>